nixos/modules/system/boot/plymouth.nix

```
@@ -16,14 +15,37 @@ let
     osVersion = config.system.nixos.release;
   };
 
+  plymouthLogos = pkgs.runCommand "plymouth-logos" { inherit (cfg) logo; } ''
+    mkdir -p $out
+
+    # For themes that are compiled with PLYMOUTH_LOGO_FILE
+    mkdir -p $out/etc/plymouth
+    ln -s $logo $out/etc/plymouth/logo.png
```

Using Nix value directly is IMO clearer than passing it to environment as a middleman:

```suggestion
    ln -s "${cfg.logo}" "$out/etc/plymouth/logo.png"
```